File: frontend/src/app/common/model/unit-type.ts
export enum UnitType {
    UNIT = 'UNIT',
    OZ = 'OZ',
    MWH = 'MWH',
    LBS = 'LBS',
    T = 'T',
    KG = 'KG',
    MTU = 'MTU',
    BU = 'BU',
    CWT = 'CWT',
    GAL = 'GAL',
    THM = 'THM',
    BBL = 'BBL',
    MMBTU = 'MMBTU',
    DOZEN = 'DOZEN',
    THOUSAND_BOARD_FEET = 'THOUSAND_BOARD_FEET'
  }
  
  export const UnitTypeLabels: { [key in UnitType]: string } = {
    [UnitType.UNIT]: 'unit',
    [UnitType.OZ]: 'oz',
    [UnitType.MWH]: 'MWh',
    [UnitType.LBS]: 'lbs',
    [UnitType.T]: 't',
    [UnitType.KG]: 'kg',
    [UnitType.MTU]: 'mtu',
    [UnitType.BU]: 'bu',
    [UnitType.CWT]: 'cwt',
    [UnitType.GAL]: 'gal',
    [UnitType.THM]: 'thm',
    [UnitType.BBL]: 'bbl',
    [UnitType.MMBTU]: 'MMBtu',
    [UnitType.DOZEN]: 'dz',
    [UnitType.THOUSAND_BOARD_FEET]: 'MBF'
};
  

File: backend/common/src/main/java/pl/muybien/enumeration/UnitType.java
package pl.muybien.enumeration;

public enum UnitType {
    UNIT,   // For generic units
    OZ,
    MWH,
    LBS,
    T,
    KG,
    MTU,
    BU,
    CWT,
    GAL,
    THM,
    BBL,
    MMBTU,
    DOZEN,
    THOUSAND_BOARD_FEET;

    public static UnitType extractUnit(String unit) {
        for (UnitType type : UnitType.values()) {
            if (unit.toUpperCase().contains(type.name())) return type;
        }
        return null;
    }
}